src/crypto/crypto_context.cc

```diff
@@ -223,10 +227,8 @@ X509_STORE* NewRootCertStore() {
   if (per_process::cli_options->ssl_openssl_cert_store) {
     X509_STORE_set_default_paths(store);
   } else {
-    for (X509* cert : root_certs_vector) {
-      X509_up_ref(cert);
```

Comment:
This was a referencing counting error in the existing code. OpenSSL's documentation for `X509_STORE_add_cert` states that "The added object's reference count is incremented by one, hence the caller retains ownership of the object and needs to free it when it is no longer needed.".